### ISSUE TITLE
Add support for request timeouts

### ISSUE THREAD

**Issue by countrodrigo** [nlf/riakpbc#69]
There doesn't seem to be to set timeouts on a per request basis. Did I read the code wrong?

**Comment by nlf** [comment]
You're absolutely correct, there isn't currently a per-request timeout.. I'll see what I can come up with.

**Comment by countrodrigo** [comment]
Oh hey, so we actually implemented this in another library, but I think your library already supports it natively. Since you are using the 1.4 protobuf spec, you can already pass in a timeout variable in the request and Riak will implement the timeout itself.

**Comment by countrodrigo** [comment]
That said, it would still be nice to be able to timeout requests that are getting hung up along the way for whatever reason, client side.

**Comment by nlf** [comment]
Yup, you can certainly pass timeout values to the commands that support them. But I agree, a client side timeout like we have for connecting would be nice.

**Comment by nlf** [comment]
I think I'm going to postpone this.. I want to get the 2.0 release out, and this will require more changes than I can manage in a short timeframe.
